logos/logos-workernode/logos_worker_node/host_ram.py

```
@@ -0,0 +1,107 @@
+        for line in Path(f"/proc/{pid}/status").read_text().splitlines():
+            if line.startswith("VmRSS:"):
+                parts = line.split()
+                if len(parts) >= 2:
+                    return float(parts[1]) / 1024.0
+    except (FileNotFoundError, PermissionError, OSError, ValueError):
+        return 0.0
+    return 0.0
+
+
+def walk_process_tree(root_pid: int, max_pids: int = 256) -> set[int]:
+    """Return {root_pid} ∪ all descendants discovered via /proc/<pid>/task/*/children.
```

Ruff (0.15.14)

[warning] Docstring contains ambiguous `∪` (UNION). Did you mean `U` (LATIN CAPITAL LETTER U)?

(RUF002)